### PR TITLE
[trivial] Tweak `std.meta` docs

### PR DESCRIPTION
**Pull request by ntrel**
* Replace uses of *list* with *sequence*.
* `S[n]` is not the nth element (due to zero based indexing).
* Add links, formatting.

### CODE REVIEW

File: std/meta.d [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
- * Templates to manipulate template argument lists (also known as type lists).
+ * Templates to manipulate
+ * $(DDSUBLINK spec/template, variadic-templates, template parameter sequences)
+ * (also known as $(I alias sequences)).
  *
- * Some operations on alias sequences are built in to the language,
- * such as TL[$(I n)] which gets the $(I n)th type from the
- * alias sequence. TL[$(I lwr) .. $(I upr)] returns a new type
- * list that is a slice of the old one.
+ * Some operations on alias sequences are built into the language -
```
Comment (JackStouffer) [inline std/meta.d]: The comma was correct. `-` is not the correct way to make a sub-clause of a sentence.

File: std/meta.d [hunk 2]
```diff
@@ -758,7 +762,7 @@ template MostDerived(T, TList...)
 }
 
 /**
- * Returns the `AliasSeq` TList with the types sorted so that the most
+ * Returns an `AliasSeq` with the elements of TList sorted so that the most
```
Comment (JackStouffer) [inline std/meta.d]: Since you're modifying this line anyway, might as well change

    `AliasSeq`

to

    $(LREF AliasSeq)

Same throughout.

**Comment by ntrel** [comment]
I restored the comma, I've not made the LREF change. I changed a link title to match https://github.com/dlang/dlang.org/pull/2204.

**Comment by JackStouffer** [comment]
>I've not made the LREF change

Any particular reason?

**Comment by ntrel** [comment]
@JackStouffer Just that I didn't want to get into another change unless it added significant value. Merge please? ;-)

**Pull request merged**